packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -270,7 +270,7 @@ export function useAsyncList<T, C = string>(options: AsyncListOptions<T, C>): As
     sort(sortDescriptor: SortDescriptor) {
       dispatchFetch({type: 'sorting', sortDescriptor}, sort || load);
     },
-    ...createListActions(options, fn => {
+    ...createListActions({...options, getKey}, fn => {
```

Comment:
This is get the default `getKey` properly passed to `useListData` if `getKey` isn't defined by the user